athena-hbase/src/main/java/com/amazonaws/athena/connectors/hbase/HbaseConnectionFactory.java

```diff
@@ -113,6 +123,40 @@ private Connection createConnection(String host, String masterPort, String zooke
                 logger.info("createConnection: applying client config {}:{}", nextConfig.getKey(), nextConfig.getValue());
                 config.set(nextConfig.getKey(), nextConfig.getValue());
             }
+
+            Map<String, String> configOptions = System.getenv();
+            boolean kerberosAuthEnabled = configOptions.get(KERBEROS_AUTH_ENABLED) != null && "true".equalsIgnoreCase(configOptions.get(KERBEROS_AUTH_ENABLED));
+            logger.info("Kerberos Authentication Enabled: " + kerberosAuthEnabled);
+            if (kerberosAuthEnabled) {
+                String keytabLocation = null;
+                config.set("hbase.rpc.protection", configOptions.get(HBASE_RPC_PROTECTION));
+                logger.info("hbase.rpc.protection: " + config.get("hbase.rpc.protection"));
+                String s3uri = configOptions.get(KERBEROS_CONFIG_FILES_S3_REFERENCE);
+                if (StringUtils.isNotBlank(s3uri)) {
+                    try {
+                        Path tempDir = copyConfigFilesFromS3ToTempFolder(configOptions);
+                        logger.debug("tempDir: " + tempDir);
+                        keytabLocation = tempDir + File.separator + "hbase.keytab";
+                        System.setProperty("java.security.krb5.conf", tempDir + File.separator + "krb5.conf");
+                        logger.debug("krb5.conf location: " + tempDir + File.separator + "krb5.conf");
+                    }
+                    catch (Exception e) {
+                        throw new RuntimeException("Exception while copying config files from S3 to temp folder: " + e);
```

Comment:
Still not using the right connector; see [`public RuntimeException(String message,Throwable cause)`](https://docs.oracle.com/javase/8/docs/api/java/lang/RuntimeException.html#RuntimeException-java.lang.String-java.lang.Throwable-)

Reply:
Hi @AbdulR3hman, Updated code as per your suggestion.